docs/faq/DesignDecisions.md

```diff
@@ -0,0 +1,75 @@
+# Redux FAQ: Design Decisions
+
+## Table of Contents
+
+- [Why doesn't Redux pass the state and action to subscribers?](#does-not-pass-state-action-to-subscribers) 
+- [Why doesn't Redux support using classes for actions and reducers?](#does-not-support-classes) 
+- [Why does the middleware signature use currying?](#why-currying)
+- [Why does applyMiddleware use a closure for dispatch?](#closure-dispatch)
+- [Can you please change combineReducers to support nested state trees?](#combineReducers-limitations)
+- [Why doesn't mapDispatchToProps allow use of return values from getState() or mapStateToProps()?](#no-asynch-in-mapDispatchToProps)
+
+
+## Design Decisions
+
+<a id="does-not-pass-state-action-to-subscribers"></a>
+### Why doesn't Redux pass the state and action to subscribers?
+Subscribers are intended to respond to the state value itself, not the action. Updates to the state are not always processed synchronously, because Redux sometimes processes actions in batches to optimize performance and repeated re-rendering. The intended guarantee is that Redux eventually calls all subscribers with the most recent state, but not that it always calls each subscriber for each action. The store state is available in the subscriber simply by calling store.getState(). The action cannot be made available in the subsribers without breaking the way that actions are batched. 
```

Comment:
There is no batching in Redux natively. However, you can add that functionality. I might reword it to say "because libraries can change Redux to process actions in batches".